Building CoreLib project and setting global properties can result in races / multiple builds

See comments here: https://github.com/dotnet/runtime/pull/38196/files#r445079996

We need to be very careful not to specify global properties which aren't necessary.

TargetFramework should not be passed at all and should be suppressed, assuming it is not passed down as part of the corelib build normally.

Configuration should only be passed if different than the Configuration of the current project.

For Configuration you could decide to use the Configuration metadata instead of the AdditionalProperties metadata but that is

Seems like I just hit this locally:

```
  System.Private.CoreLib -> C:\git\runtime4\artifacts\bin\coreclr\Windows_NT.x64.Release\IL\System.Private.CoreLib.dll
  System.Private.CoreLib -> C:\git\runtime4\artifacts\bin\coreclr\Windows_NT.x64.Release\IL\System.Private.CoreLib.dll
C:\git\runtime4\.dotnet\sdk\5.0.100-preview.6.20310.4\Microsoft.Common.CurrentVersion.targets(4254,5): error MSB3026: Could not copy "C:\git\runtime4\artifacts\obj\coreclr\System.Private.CoreLib\x64\Release\System.Private.CoreLib.pdb" to "C:\git\runtime4\artifacts\bin\coreclr\Windows_NT.x64.Release\/IL/System.Private.CoreLib.pdb". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\git\runtime4\artifacts\bin\coreclr\Windows_NT.x64.Release\IL\System.Private.CoreLib.pdb' because it is being used by another process.  [C:\git\runtime4\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

--- a/src/libraries/Directory.Build.targets
+++ b/src/libraries/Directory.Build.targets
@@ -182,12 +182,12 @@
     <ItemGroup>
       <ProjectReference>
         <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
-        <UndefineProperties Condition="'%(Filename)' == 'System.Private.CoreLib'">$(UndefineProperties);TargetFramework</UndefineProperties>
-        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
-                                     '$(RuntimeFlavor)' == 'CoreCLR' and
+      </ProjectReference>
+      <ProjectReference Condition="'%(Filename)' == 'System.Private.CoreLib'">
+        <UndefineProperties>$(UndefineProperties);TargetFramework</UndefineProperties>
+        <SetConfiguration Condition="'$(RuntimeFlavor)' == 'CoreCLR' and
                                      '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</SetConfiguration>
-        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
-                                     '$(RuntimeFlavor)' == 'Mono' and
+        <SetConfiguration Condition="'$(RuntimeFlavor)' == 'Mono' and
                                      '$(Configuration)' != '$(MonoConfiguration)'">Configuration=$(MonoConfiguration)</SetConfiguration>
       </ProjectReference>
     </ItemGroup>

--- a/src/libraries/Directory.Build.targets
+++ b/src/libraries/Directory.Build.targets
@@ -182,8 +182,13 @@
     <ItemGroup>
       <ProjectReference>
         <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
-        <AdditionalProperties Condition="'%(Filename)' == 'System.Private.CoreLib' and '$(RuntimeFlavor)' == 'CoreCLR'">Configuration=$(CoreCLRConfiguration);TargetFramework=$(NetCoreAppCurrent)</AdditionalProperties>
-        <AdditionalProperties Condition="'%(Filename)' == 'System.Private.CoreLib' and '$(RuntimeFlavor)' == 'Mono'">Configuration=$(MonoConfiguration);TargetFramework=$(NetCoreAppCurrent)</AdditionalProperties>
+        <UndefineProperties Condition="'%(Filename)' == 'System.Private.CoreLib'">$(UndefineProperties);TargetFramework</UndefineProperties>
+        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
+                                     '$(RuntimeFlavor)' == 'CoreCLR' and
+                                     '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</SetConfiguration>
+        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
+                                     '$(RuntimeFlavor)' == 'Mono' and
+                                     '$(Configuration)' != '$(MonoConfiguration)'">Configuration=$(MonoConfiguration)</SetConfiguration>
       </ProjectReference>
     </ItemGroup>
   </Target>

